zc.recipe.egg_/src/zc/recipe/egg/egg.py

```diff
@@ -166,6 +159,60 @@ def install(self):
 
     update = install
 
+
+_working_set_cache = {}
+
+
+def working_set(
+    distributions,
+    eggs_dir,
+    develop_eggs_dir,
+    offline,
+    newest,
+    links,
+    index,
+    allow_hosts,
+    extra=(),
+):
+    """Helper function to build a working set.
+
+    Results are cached in the `_working_set_cache` dict at module level.
+    The cache key is composed by all the arguments passed to the function.
+    """
+    orig_distributions = list(distributions)
+    cache_key = (
+        tuple(distributions),
```

Comment:
Can the order of distributions affect the generated working set? If not then definitely the list should be sorted.